modules/behavior/draw_way.js

```diff
@@ -423,74 +423,53 @@ export function behaviorDrawWay(context, wayID, mode, startGraph) {
         return 'generic';
     }
 
-    /** see PR #8671 */
+    /** see PR #8671 and #9340 */
     function followMode() {
         if (_didResolveTempEdit) return;
 
         try {
-
-            // get the last 2 added nodes.
-            // check if they are both part of only oneway (the same one)
-            // check if the ways that they're part of are the same way
-            // find index of the last two nodes, to determine the direction to travel around the existing way
-            // add the next node to the way we are drawing
-
             // if we're drawing an area, the first node = last node.
-            const isDrawingArea = _origWay.nodes[0] === _origWay.nodes.slice(-1)[0];
+            const isDrawingArea = _origWay.nodes.length > 1 && _origWay.nodes[0] === _origWay.nodes[_origWay.nodes.length - 1];
 
-            const [secondLastNodeId, lastNodeId] = _origWay.nodes.slice(isDrawingArea ? -3 : -2);
+            const secondLastNodeId = _origWay.nodes[_origWay.nodes.length + (isDrawingArea ? -3 : -2)];
```

Comment:
I would rather use `.at` but am not sure if [90%](https://caniuse.com/mdn-javascript_builtins_array_at) browser support is good enough.
```suggestion
            const secondLastNodeId = _origWay.nodes.at (isDrawingArea ? -3 : -2);
```

Reply:
> I would rather use `.at` but am not sure if [90%](https://caniuse.com/mdn-javascript_builtins_array_at) browser support is good enough.

90% is not quite usable enough for iD, I'd say. :thinking: But what was the reason for getting rid of the previous `.slice(-n)` approach? Was it because the nodes array can be shorter than 2 (or 3 for areas) nodes in the corner case when one just starts to draw a new line/area? To fix that, we could just pad the array before slicing it, e.g.:

```javascript
const [secondLastNodeId, lastNodeId] = Array(1).concat(_origWay.nodes).slice(isDrawingArea ? -3 : -2);
```

Reply:
> nodes in the corner case when one just starts to draw a new line/area

Exactly.

I will keep it this way if you don't mind, even tho padding is a great solution as well.